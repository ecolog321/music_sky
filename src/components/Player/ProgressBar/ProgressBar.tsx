import styles from './progressBar.module.css'

import { ChangeEvent, FC } from "react";

type Props={
    max:number,
    value:number,
    step:number,
    onChange:(value: ChangeEvent<HTMLInputElement>) => void;
}

export const ProgressBar: FC<Props>=({ max, value, step, onChange }) =>{
  return (
    <input
    name='progress'
      className={styles.styledProgressInput} 
      type="range" 
      min="0" 
      max={max} 
      value={value} 
      step={step} 
      onChange={onChange}
    />
  );
}
