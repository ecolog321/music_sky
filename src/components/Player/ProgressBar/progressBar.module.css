.bar__player_progress {
    width: 100%;
    height: 5px;
    background: #2e2e2e;
    color: blueviolet;
  }