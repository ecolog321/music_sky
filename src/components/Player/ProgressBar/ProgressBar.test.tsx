import { ChangeEvent, useRef } from "react";
import { ProgressBar } from "./ProgressBar";
import renderer from "react-test-renderer";

describe("Component ProgressBar", () => {
  it("render progressbar", () => {
    const props = {
      max: 100,
      value: 10,
      step: 0.5,
      onChange: ()=>{}
    };
    const tree = renderer
      .create(
        <ProgressBar
          max={props.max}
          value={props.value}
          step={props.step}
          onChange={props.onChange}
        ></ProgressBar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
