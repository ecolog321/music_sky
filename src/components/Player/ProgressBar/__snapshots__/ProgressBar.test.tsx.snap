// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ProgressBar render progressbar 1`] = `
<input
  className="styledProgressInput"
  max={100}
  min="0"
  onChange={[Function]}
  step={0.5}
  type="range"
  value={10}
/>
`;
